weblab/entities/forms.py

```diff
@@ -180,3 +180,51 @@ class FileUploadForm(forms.ModelForm):
     class Meta:
         model = EntityFile
         fields = ['upload']
+
+
+class ModelGroupForm(UserKwargModelFormMixin, forms.ModelForm):
+    """Used for creating a new model group."""
+    visibility = forms.ChoiceField(
+        choices=visibility.CHOICES,
+        help_text=visibility.HELP_TEXT.replace('\n', '<br />'),
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Only show models I can can see
+        self.current_title = None
+        self.fields['models'].queryset = ModelEntity.objects.visible_to_user(self.user)
+
+        # Save current details if we have them
+        if 'instance' in kwargs and kwargs['instance']:
+            self.current_title = kwargs['instance'].title
+            self.fields['visibility'].disabled = not kwargs['instance'].is_visibility_editable_by(self.user)
+
+            # make sure currently selected models are not filtered out even if they are not visible to the current user
+            current_models_ids = [model.id for model in kwargs['instance'].models.all()]
+            self.fields['models'].queryset |= ModelEntity.objects.filter(id__in=current_models_ids)
+
+        if not self.user.has_perm('entities.moderator'):
+            self.fields['visibility'].choices.remove((
+                visibility.Visibility.MODERATED, 'Moderated')
+            )
+
+    class Meta:
+        model = ModelGroup
+        fields = ['title', 'visibility', 'models']
+
+    def clean_title(self):
+        title = self.cleaned_data['title']
+        if title != self.current_title and self._meta.model.objects.filter(title=title, author=self.user).exists():
```

Comment:
Do we need to disallow this? Probably doesn't make sense for a user to make groups with same name, but we can still have e.g. groups with the same name but a different owner, or groups with the same name but different capitalisation, or extra spaces etc. So maybe no need to restrict this one?

Reply:
> ...  but we can still have e.g. groups with the same name but a different owner ...
That's what this does. It checks that if you'ved changed the title it's not the same as one of you own models